### PR TITLE
Gh 2589/distribution screen

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #2589 

simulator_screenshot_3695FF59-E567-4D29-B490-C8BCFB836C7A

### CODE REVIEW

**Comment by biafra23** [comment]
Looks very good. Except for the line between the navbar and the content. Which is not in the design. 
image

Is it possible to get the back button and not line between navbar and content?
`ViewControllerFactory.makeTransparentNavigationBar(self)` will eliminate the line, but it also removes the Back button.

**Review by biafra23 — APPROVED** [review]
LGTM. How do we remove the divider between navbar and content?

File: Multisig/UI/ClaimToken/LegalDisclaimerViewController.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+//
+
+import UIKit
+
+class LegalDisclaimerViewController: UIViewController {
+    @IBOutlet private weak var agreeButton: UIButton!
+    @IBOutlet private weak var textLabel: UILabel!
+
+    var onAgree: (() -> ())?
+    override func viewDidLoad() {
```
Comment (elgatovital) [inline Multisig/UI/ClaimToken/LegalDisclaimerViewController.swift:17]: add space btw func and property

File: Multisig/UI/ClaimToken/TokenDistributionViewController.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+class TokenDistributionViewController: UIViewController {
+
+    @IBOutlet private weak var titleLabel: UILabel!
+    @IBOutlet private weak var descriptionLabel: UILabel!
+    @IBOutlet private weak var nextButton: UIButton!
+    @IBOutlet weak var distributionView: BorderedCheveronButton!
+
+    private var onNext: (() -> ())?
+    private var stepNumber: Int = 1
+    private var maxSteps: Int = 3
```
Comment (DmitryBespalov) [inline Multisig/UI/ClaimToken/TokenDistributionViewController.swift]: change it to 4

File: Multisig/UI/ClaimToken/TokenDistributionViewController.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        self.init(namedClass: TokenDistributionViewController.self)
+        self.stepNumber = stepNumber
+        self.maxSteps = maxSteps
+        self.onNext = onNext
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        distributionView.set("Distribution details") {
+            //TODO: Show distribution details
```
Comment (DmitryBespalov) [inline Multisig/UI/ClaimToken/TokenDistributionViewController.swift:36]: probably should call a callback so that flow can open the details

Comment (MouazAlzahabi) [reply Multisig/UI/ClaimToken/TokenDistributionViewController.swift:36]: presenting the details will be done from this point when the screen is ready, no need for callback, no?

File: Multisig/UI/ClaimToken/TokenDistributionViewController.swift (hunk 2 above)
Comment (MouazAlzahabi) [reply Multisig/UI/ClaimToken/TokenDistributionViewController.swift]: updated, however it was being initialized with correct value in the init method

File: Multisig/UI/ClaimToken/TokenDistributionViewController.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
+//
+//  TokenDistributionViewController.swift
+//  Multisig
+//
+//  Created by Mouaz on 9/5/22.
+//  Copyright © 2022 Gnosis Ltd. All rights reserved.
```
Comment (elgatovital) [inline Multisig/UI/ClaimToken/TokenDistributionViewController.swift:6]: we should update the copyright

Comment (MouazAlzahabi) [reply Multisig/UI/ClaimToken/TokenDistributionViewController.swift:6]: we have this all over the app, I will change it later with the full rebrand

**Pull request merged**